internal/api/middleware/timer.go

```diff
@@ -0,0 +1,43 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/dogstatsd"
+)
+
+func getHist(r *http.Request, t *dogstatsd.Timing) metrics.Histogram {
```

Comment:
As discussed offline, this function does too much. I'd suggest writing a single responsibility function that gets and formats the stable endpoint id, and move the rest of the responsibilities (getting the request method, adding the metric labels) to the middleware func.